src/pages/trackedEntityTypes/form/TrackedEntityTypeFormSchema.ts

```diff
@@ -15,25 +19,52 @@ const trackedEntityTypeBaseSchema = z.object({
             icon: z.string().optional(),
         })
         .default({}),
-    allowAuditLog: z.boolean().optional(),
-    minAttributesRequiredToSearch: z.number().optional(),
-    maxTeiCountToReturn: z.number().optional(),
+    allowAuditLog: z.boolean().default(false),
+    minAttributesRequiredToSearch: z
+        .number()
+        .min(
+            0,
+            'Minimum number of attributes required to search must be 0 or greater'
+        )
+        .default(1),
+    maxTeiCountToReturn: z
+        .number()
+        .min(
+            0,
+            'Maximum number of tracked entity instances to return must be 0 or greater'
+        )
+        .default(0),
     featureType: z.string().trim().optional(),
-    attributeValues: z
+    trackedEntityTypeAttributes: z
         .array(
             z.object({
-                value: z.string(),
-                attribute: z.object({
+                mandatory: z.boolean().default(false),
+                searchable: z.boolean().default(false),
+                displayInList: z.boolean().default(false),
+                trackedEntityAttribute: z.object({
                     id: z.string(),
+                    displayName: z.string(),
                 }),
```

Comment:
I just learned that we use `modelReference` when we only need the ID of a related model object, and extend it with' .extend() ' if we need additional fields. Hope that is helpful for you as well, @edvinstava.
```suggestion
                trackedEntityAttribute: modelReference.extend({
                       displayName: z.string(),
                })
```